<!-- sct-test-summary -->
## :x: Test Summary: FAILED

### :white_check_mark: Precommit: PASSED

| Total | Passed | Failed | Skipped |
|-------|--------|--------|---------|
| 54 | 18 | 0 | 36 |


### :x: Tests: FAILED

| Total | Passed | Failed | Errors | Skipped |
|-------|--------|--------|--------|---------|
| 4607 | 4552 | 3 | 0 | 52 |

### Failed Tests

#### `integration.test_config_get_version_based_on_conf.test_unified_package`
**Type:** FAILURE
**Message:** `FileNotFoundError: Unified package not found or failed to download: https://downloads.scylladb.com/unstable/scylla/master/relocatable/2023-11-13T03:04:27Z/scylla-unified-5.5.0~dev-0.20231113.7b08886e8dd8.x86_64.tar.gz. The URL may not exist or is not accessible.`
<details><summary>Traceback</summary>

```
self = SCTConfiguration(perf_simple_query_extra_command=None, force_run_iotune=None, data_volume_disk_throughput=None, multi_...d_regions=['eu-west-1', 'eu-west-2', 'eu-west-3', 'us-west-2', 'us-east-1', 'us-east-2', 'eu-north-1', 'eu-central-1'])

    def get_version_based_on_conf(self):
        """
        figure out which version and if it's enterprise version
        base on configuration only, before nodes are up and running
        so test configuration can set up things which need to happen
        before nodes are up
    
        this is information is cached on the SCTConfiguration object
        :return: tuple - (scylla_version, is_enterprise)
        """
        backend = self.get("cluster_backend")
        scylla_version = None
        _is_enterprise = False
    
        if self.get("db_type") in ("cassandra",):
            return scylla_version, _is_enterprise
    
        if unified_package := self.get("unified_package"):
            with tempfile.TemporaryDirectory() as tmpdirname:
                try:
>                   LOCALRUNNER.run(
                        shell_script_cmd(f"""
                        cd {tmpdirname}
                        {curl_with_retry(unified_package, output="./unified_package.tar.gz", follow_redirects=True, fail_early=True)}
                        tar xvfz ./unified_package.tar.gz
                        """),
                        verbose=False,
                    )

sdcm/sct_config.py:4514: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sdcm/remote/local_cmd_runner.py:100: in run
    result = _run()
             ^^^^^^
sdcm/utils/decorators.py:74: in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
sdcm/remote/local_cmd_runner.py:90: in _run
    result = self.connection.local(**command_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/local/lib/python3.14/site-packages/fabric/connection.py:870: in local
    return super().run(*args, **kwargs)

```
</details>

#### `integration.test_external_backtrace_service.test_decode_via_external_service_returns_symbols`
**Type:** FAILURE
**Message:** `requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://backtrace.scylladb.com/api/backtrace`
<details><summary>Traceback</summary>

```
node = <unit_tests.lib.fake_cluster.DummyNode object at 0x739a6aa77b60>

    def test_decode_via_external_service_returns_symbols(node):
        """Verify _decode_via_external_service returns decoded symbols for a known build."""
>       result = node._decode_via_external_service(KNOWN_BUILD_ID, SAMPLE_BACKTRACE_ADDRESSES)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

unit_tests/integration/test_external_backtrace_service.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sdcm/cluster.py:2035: in _decode_via_external_service
    response.raise_for_status()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [404]>

    def raise_for_status(self):
        """Raises :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ""
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
            # isn't utf-8, we fall back to iso-8859-1 for all other
            # encodings. (See PR #3538)
            try:
                reason = self.reason.decode("utf-8")
            except UnicodeDecodeError:
                reason = self.reason.decode("iso-8859-1")
        else:
            reason = self.reason
    
        if 400 <= self.status_code < 500:
            http_error_msg = (
                f"{self.status_code} Client Error: {reason} for url: {self.url}"
            )
    
        elif 500 <= self.status_code < 600:
            http_error_msg = (
                f"{self.status_code} Server Error: {reason} for url: {self.url}"
            )
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://backtrace.scylladb.com/api/backtrace

/usr/local/lib/python3.14/site-packages/requests/mod
```
</details>

#### `integration.test_version_utils.test_get_specific_tag_of_docker_image[scylladb/scylla-enterprise-nightly]`
**Type:** FAILURE
**Message:** `requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://s3.amazonaws.com/downloads.scylladb.com/unstable/scylla-enterprise/enterprise/relocatable/latest/00-Build.txt`
<details><summary>Traceback</summary>

```
docker_repo = 'scylladb/scylla-enterprise-nightly'

    @pytest.mark.need_network
    @pytest.mark.integration
    @pytest.mark.parametrize("docker_repo", ["scylladb/scylla-nightly", "scylladb/scylla-enterprise-nightly"])
    def test_get_specific_tag_of_docker_image(docker_repo):
>       assert get_specific_tag_of_docker_image(docker_repo=docker_repo) != "latest"
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

unit_tests/integration/test_version_utils.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sdcm/utils/version_utils.py:688: in get_specific_tag_of_docker_image
    res.raise_for_status()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [404]>

    def raise_for_status(self):
        """Raises :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ""
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
            # isn't utf-8, we fall back to iso-8859-1 for all other
            # encodings. (See PR #3538)
            try:
                reason = self.reason.decode("utf-8")
            except UnicodeDecodeError:
                reason = self.reason.decode("iso-8859-1")
        else:
            reason = self.reason
    
        if 400 <= self.status_code < 500:
            http_error_msg = (
                f"{self.status_code} Client Error: {reason} for url: {self.url}"
            )
    
        elif 500 <= self.status_code < 600:
            http_error_msg = (
                f"{self.status_code} Server Error: {reason} for url: {self.url}"
            )
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://s3.amazonaws.com/downloads.scylladb.com/unstable/scylla-enterprise/enterp
```
</details>

[Full build log](https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-15670/2/)

--- a/sdcm/keystore.py
+++ b/sdcm/keystore.py
@@ -43,6 +43,10 @@ KEYSTORE_S3_BUCKET = "scylla-qa-keystore"
 # client must always be pinned explicitly.
 KEYSTORE_SM_REGION = "us-east-1"
 
+# Namespace the mirrored keystore entries live under in Secrets Manager, so a
+# keystore file `foo.json` maps to the secret `sct/foo.json`.
+KEYSTORE_SM_PREFIX = "sct/"
+
 SSHKey = namedtuple("SSHKey", ["name", "public_key", "private_key"])
 
 BOTO3_CLIENT_CREATION_LOCK = threading.Lock()
@@ -99,7 +103,7 @@ class KeyStore:
         self._cache: dict[str, bytes] = {}
         self._cache_lock = threading.Lock()
         self._backend = backend or os.environ.get("SCT_KEYSTORE_BACKEND") or "secretsmanager"
-        self._sm_prefix = os.environ.get("SCT_KEYSTORE_SM_PREFIX") or "sct/"
+        self._sm_prefix = os.environ.get("SCT_KEYSTORE_SM_PREFIX") or KEYSTORE_SM_PREFIX
         self._sm_region = os.environ.get("SCT_KEYSTORE_SM_REGION") or KEYSTORE_SM_REGION
 
     @property
@@ -431,8 +435,17 @@ class KeyStore:
             self.download_file(filename=key, dest_filename=path)
             os.chmod(path=path, mode=permissions)
             if self._backend == "secretsmanager":
-                with open(f"{path}.version", "w", encoding="utf-8") as vf:
+                with open(version_path, "w", encoding="utf-8") as vf:
                     vf.write(remote_version)
+
+        if self._backend == "secretsmanager":
+            # Match the key's permissions: the sidecar lands in the same
+            # directory (often ~/.ssh, where ssh rejects group/world readable
+            # content) and would otherwise inherit the umask. Enforced on every
+            # sync rather than only after a download, because a sidecar whose
+            # version is still current is never rewritten -- one left at 0o664
+            # by an earlier sync would keep those permissions forever.
+            os.chmod(path=version_path, mode=permissions)
 
     def sync(self, keys, local_path, permissions=0o777):
         """Syncs the local and remote copies from the configured backend."""

--- a/unit_tests/integration/test_aws_services.py
+++ b/unit_tests/integration/test_aws_services.py
@@ -12,7 +12,7 @@ import boto3
 os.environ["MOTO_AMIS_PATH"] = str(Path(__file__).parents[1] / "test_data" / "mocked_ami_data.json")
 from moto.server import ThreadedMotoServer
 
-from sdcm.keystore import KeyStore
+from sdcm.keystore import KEYSTORE_S3_BUCKET, KEYSTORE_SM_PREFIX, KEYSTORE_SM_REGION, KeyStore
 from sdcm.utils.aws_region import AwsRegion
 from sdcm.sct_provision.common.layout import SCTProvisionLayout, create_sct_configuration
 from sdcm.utils.common import get_scylla_ami_versions
@@ -29,7 +29,7 @@ AWS_REGION = "us-east-1"
 @pytest.fixture(scope="session", autouse=True)
 def fixture_get_real_keys():
     KeyStore().sync(
-        keys=["scylla-qa-ec2", "scylla-test", "scylla_test_id_ed25519", "scylla_test_id_ed25519.pub"],
+        keys=["scylla_test_id_ed25519", "scylla_test_id_ed25519.pub"],
         local_path=Path("~/.ssh/").expanduser(),
         permissions=0o0600,
     )
@@ -55,17 +55,27 @@ def moto_server():
 
 @pytest.fixture(scope="module", autouse=True)
 def keystore_configure(moto_server):
-    s3 = boto3.resource(service_name="s3", region_name=AWS_REGION, endpoint_url=moto_server)
+    entries = {
+        "gcp-sct-project-1.json": b"{}",
+        "aws_images_role.json": (
+            b'{"role_arn": "arn:aws:iam::123456789012:role/role-name", "role_session_name": "role-session-name"}'
+        ),
+    }
+    for file in ("scylla_test_id_ed25519", "scylla_test_id_ed25519.pub"):
+        entries[file] = (Path("~/.ssh").expanduser() / file).read_bytes()
 
-    bucket = s3.Bucket("scylla-qa-keystore")
+    s3 = boto3.resource(service_name="s3", region_name=AWS_REGION, endpoint_url=moto_server)
+    bucket = s3.Bucket(KEYSTORE_S3_BUCKET)
     bucket.create()
-    bucket.put_object(Key="gcp-sct-project-1.json", Body=b"{}")
-    bucket.put_object(
-        Key="aws_images_role.json",
-        Body=b'{"role_arn": "arn:aws:iam::123456789012:role/role-name", "role_session_name": "role-session-name"}',
-    )
-    for file in ("scylla-qa-ec2", "scylla-test", "scylla_test_id_ed25519", "scylla_test_id_ed25519.pub"):
-        bucket.upload_file(Filename=str(Path("~/.ssh").expanduser() / file), Key=file)
+    for key, body in entries.items():
+        bucket.put_object(Key=key, Body=body)
+
+    # KeyStore reads from Secrets Manager by default, so mirror the same entries
+    # there under the `sct/` prefix - seeding only the bucket would leave every
+    # lookup failing with ResourceNotFoundException.
+    sm = boto3.client("secretsmanager", region_name=KEYSTORE_SM_REGION, endpoint_url=moto_server)
+    for key, body in entries.items():
+        sm.create_secret(Name=f"{KEYSTORE_SM_PREFIX}{key}", SecretBinary=body)
 
 
 @pytest.fixture(scope="module")
@@ -93,7 +103,7 @@ def test_02_keystore_sync(tmp_path) -> None:
     Validate the sync is working and setting right permissions.
     """
     k = KeyStore()
-    k.sync(keys=["scylla-qa-ec2", "scylla-test", "scylla_test_id_ed25519"], local_path=tmp_path, permissions=0o0600)
+    k.sync(keys=["scylla_test_id_ed25519", "scylla_test_id_ed25519.pub"], local_path=tmp_path, permissions=0o0600)
 
     for file in tmp_path.iterdir():
         assert stat.S_IMODE(file.stat().st_mode) == 0o600

--- a/unit_tests/integration/test_aws_services.py
+++ b/unit_tests/integration/test_aws_services.py
@@ -9,7 +9,7 @@ import boto3
 # we need to set the environment variable before importing moto
 # otherwise it won't pick it up
 
-os.environ["MOTO_AMIS_PATH"] = str(Path(__file__).parent / "test_data" / "mocked_ami_data.json")
+os.environ["MOTO_AMIS_PATH"] = str(Path(__file__).parents[1] / "test_data" / "mocked_ami_data.json")
 from moto.server import ThreadedMotoServer
 
 from sdcm.keystore import KeyStore

--- a/unit_tests/integration/test_sync.py
+++ b/unit_tests/integration/test_sync.py
@@ -23,7 +23,7 @@ def test_multiple_sync_on_lots_of_files():
     if not is_using_aws_mock():
         key_store = KeyStore()
         key_store.sync(
-            keys=["scylla-qa-ec2", "scylla-test", "scylla_test_id_ed25519"] * 20,
+            keys=["scylla_test_id_ed25519", "scylla_test_id_ed25519.pub"] * 30,
             local_path=Path("~/.ssh/").expanduser(),
             permissions=0o0600,
         )

--- a/unit_tests/unit/test_keystore.py
+++ b/unit_tests/unit/test_keystore.py
@@ -22,6 +22,7 @@ import io
 import json
 import logging
 import os
+import stat
 import time
 from concurrent.futures.thread import ThreadPoolExecutor
 from unittest.mock import MagicMock, PropertyMock, patch
@@ -975,6 +976,24 @@ class TestSecretsManagerBackend:
             assert fh.read() == FAKE_SSH_PRIVATE_KEY
         # Version sidecar written alongside the key
         assert (tmp_path / "scylla_test_id_ed25519.version").exists()
+
+    def test_sync_from_sm_applies_permissions_to_sidecar(self, sm_ks, tmp_path):
+        """The sidecar must not inherit the umask - ssh rejects a group/world readable ~/.ssh."""
+        sm_ks.sync(["scylla_test_id_ed25519"], str(tmp_path), 0o600)
+        for name in ("scylla_test_id_ed25519", "scylla_test_id_ed25519.version"):
+            assert stat.S_IMODE((tmp_path / name).stat().st_mode) == 0o600, f"{name} has wrong permissions"
+
+    def test_sync_from_sm_repairs_sidecar_permissions_on_cache_hit(self, sm_ks, tmp_path):
+        """A sidecar left too permissive by an earlier sync is repaired without a re-download."""
+        sm_ks.get_obj_if_needed("scylla_test_id_ed25519", str(tmp_path), 0o600)
+        sidecar = tmp_path / "scylla_test_id_ed25519.version"
+        sidecar.chmod(0o664)
+
+        # Version is unchanged, so this must fix the permissions without downloading again
+        with patch.object(sm_ks, "download_file") as mock_dl:
+            sm_ks.get_obj_if_needed("scylla_test_id_ed25519", str(tmp_path), 0o600)
+            mock_dl.assert_not_called()
+        assert stat.S_IMODE(sidecar.stat().st_mode) == 0o600
 
     def test_get_sm_version_id_returns_awscurrent(self, sm_ks):
         version_id = sm_ks.get_sm_version_id("scylla_test_id_ed25519")

